Extend hackathon page (#936)
gardener/documentation: website/community/hackathons/_index.md

However, it's important to note that these teams are not fixed, and participants have the flexibility to switch topics dynamically during the week as they see fit, allowing for a fluid and collaborative environment.

Long story short: Any topic that is agreed upon in the round of the hackathon participants is in scope for a hackathon.

## 🤝 How to Participate

There is a Slack channel called [`#hack-the-garden`](https://gardener-cloud.slack.com/archives/C0531FVMZFU) in [Gardener's Slack workspace](https://join.slack.com/t/gardener-cloud/shared_invite/zt-33c9daems-3oOorhnqOSnldZPWqGmIBw). Feel free to join and reach out! The community is always happy to welcome new faces to the event.

### Technical Prerequisites

Gardener hackathons are a great opportunity for meeting new people in the community and collaborating on interesting topics from various parts of the codebase.
Nonetheless, please be aware that you need to have a certain level of technical knowledge to be able to contribute effectively and making the event worthwhile for you as well!

**Required:**

> * Solid experience programming in [Go](https://go.dev/) and ideally already familiar developing Kubernetes controllers using [controller-runtime](https://github.com/kubernetes-sigs/controller-runtime).

Reviewer: I had to re-read `Solid experience programming in Go` multiple times until I understood it. Maybe we can rephrase it. My expectation would have been `solid experience in programming`, but then you end up with `in programming in go`...

* Familiarity with [Kubernetes](https://kubernetes.io/) concepts and architecture, as well as experience working with Kubernetes APIs and resources.
* An understanding of Gardener's architecture and components, as well as its codebase and development practices.

> * Prepare the [local setup for Gardener](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally.md) development. All required dependencies need to be downloaded before the event and your device should be capable of running Gardener locally.

Reviewer: I find this a bit hard as you do not need everything listed here. It is certainly a plus, but we have also had people without a complete understanding of the Gardener codebase. The same is also true for other points.
Maybe I just see `required` as too strict. It is certainly recommended, but we had quite a few people joining that did not tick all boxes. Still, it was good.

Reviewer 2: Agree, what about "Strongly Recommended" ?

**Optional:**
* Experience with cloud providers and their APIs, as Gardener supports multiple infrastructure providers (hyperscalers, OpenStack, bare metal, you name it).

### Organization Prerequisites

Due to logistics and available space at the event location there is a limit of 30 participants per event.

> Make sure to follow ongoing planning activities in the [`#hack-the-garden`](https://gardener-cloud.slack.com/archives/C0531FVMZFU) Slack channel and register yourself on time!

Reviewer: Not a native speaker, but I would rather register before and not directly at the deadline ;-) ...

```suggestion
Make sure to follow ongoing planning activities in the [`#hack-the-garden`](https://gardener-cloud.slack.com/archives/C0531FVMZFU) Slack channel and register yourself in time!
```

### Which topics are in scope?

In general, all topics in the ecosystem of Gardener are in scope.
This includes, but is not limited to, anything directly part of [gardener/gardener](https://github.com/gardener/gardener), extensions, infrastructure providers, documentation, testing, performance evaluations, and anything else that you can think of that is related to Gardener!

Once the planning for a hackathon event kicks off, topic proposals are collected in the [hackathon repository](https://github.com/gardener/hackathon) as issues.
These issues are discussed and presented to each other in a topic collection meeting in the round of the participants.
After the topic collection, a voting process takes place where each participant can vote for the topics they are interested in.
Based on the voting results, initial teams are formed around the most popular topics, ensuring a mix of participants from different companies to foster diversity and collaboration.
However, it's important to note that these teams are not fixed, and participants have the flexibility to switch topics dynamically during the week as they see fit, allowing for a fluid and collaborative environment.

> Long story short: Any topic that is agreed upon in the round of the hackathon participants is in scope for a hackathon.

Reviewer 3: ```suggestion
Long story short: Any topic that the hackathon participants agree on is in scope.
```
I'd suggest rewording this sentence it order to make it easier to parse and understand.
